hummingbot/connector/gateway/clob_spot/data_sources/kujira/kujira_helpers.py

```diff
@@ -0,0 +1,65 @@
+import asyncio
+import hashlib
+import traceback
+from datetime import datetime
+from typing import Any, List
+
+import jsonpickle
+
+
+def generate_hash(input: Any) -> str:
+    return generate_hashes([input])[0]
+
+
+def generate_hashes(inputs: List[Any]) -> List[str]:
+    hashes = []
+    salt = datetime.now()
+
+    for input in inputs:
+        serialized = jsonpickle.encode(input, unpicklable=True)
+        hasher = hashlib.md5()
+        target = f"{salt}{serialized}".encode("utf-8")
+        hasher.update(target)
+        hash = hasher.hexdigest()
+
+        hashes.append(hash)
+
+    return hashes
+
+
+def convert_hb_trading_pair_to_market_name(trading_pair: str) -> str:
+    return trading_pair.replace("-", "/")
+
+
+def convert_market_name_to_hb_trading_pair(market_name: str) -> str:
+    return market_name.replace("/", "-")
+
+
+def automatic_retry_with_timeout(retries=1, delay=0, timeout=None):
```

Comment:
cool decorator!!

Reply:
Hehe, thanks, it's really helpful ;)
I was thinking about creating a utils.py file so everyone could reuse it,
but then we preferred to not create extra files that would live outside of our connector for simplicity.

We have something similar on our Gateway implementation as well, but this one can cover the client too (which something is helpful to avoid disconnections or broken pipe errors).